File: snippets/rb/array/insert/insert.rb
array = [3, 2, 1]
index = 1  # 0: first item, 1: second item...
item = 100
print(array)  # [3, 2, 1]
print("\n")
array = array[0,index] + [item] + array[index,array.length()]  # ≡
print(array)  # [3, 100, 2, 1]

File: snippets/c++/array/insert/insert.cpp
#include<iostream>
#include<vector>

using namespace std;

int main() {
   int array[] = {3, 2, 1};

    cout<<"before inserting"<<endl;
    for (int i = sizeof(array) / sizeof(array[0])-1; i >= 0; i--) {
        cout << array[i];
    }
    cout << endl;
    int item = 100;
    int index = 1;
    array[index] = item;

    cout<<"after inserting"<<endl;
    for (int i = sizeof(array) / sizeof(array[0])-1; i >= 0; i--) {
        cout << array[i];
    }
    cout << endl;
   return 0;
}